src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java

```diff
@@ -15,88 +16,81 @@
 
 /**
  * An implementation class of the User Data Access Object.
- * */
-
+ */
 public class UserDaoImpl implements UserDao {
 
     public static final Logger LOGGER = LoggerFactory.getLogger(UserDaoImpl.class);
+
     @Override
-    public User findUserByUsername(String username){
-        String selectByIdQuery = "SELECT * FROM login WHERE username=?";
-        User User = null;
+    public User findUserByUsername(String username) {
+        User user = null;
 
         try (Connection connection = ConnectionHelper.getConnection();
-                  PreparedStatement preparedStatement = connection.prepareStatement(selectByIdQuery)) {
-                preparedStatement.setString(1, username);
-
-                try (ResultSet resultSet = preparedStatement.executeQuery()) {
-                    if (resultSet.next()) {
-                        User = new User();
-                        User.setId((int) resultSet.getLong("id"));
-                        User.setUsername(resultSet.getString("username"));
-                        User.setPassword(resultSet.getString("password"));
-                        User.setEntity_id(resultSet.getString("entity_id"));
-                        User.setDate_created(resultSet.getTimestamp("date_created"));
-                        User.setDate_modified(resultSet.getTimestamp("date_modified"));
-                    }
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_USER_BY_USERNAME_STATEMENT)) {
+            preparedStatement.setString(1, username);
+
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    user = new User();
+                    user.setId(resultSet.getInt("id"));
+                    user.setUsername(resultSet.getString("username"));
+                    user.setPassword(resultSet.getString("password"));
+                    user.setEntity_id(resultSet.getString("entity_id"));
+                    user.setDate_created(resultSet.getTimestamp("date_created"));
+                    user.setDate_modified(resultSet.getTimestamp("date_modified"));
                 }
+            }
         } catch (Exception e) {
-        LOGGER.error("An SQL Exception occurred." + e.getMessage());
+            LOGGER.error("An SQL Exception occurred." + e.getMessage());
         }
-       LOGGER.debug("Finding User failed.");
-        return User;
+        LOGGER.debug("Finding User failed.");
+        return user;
     }
 
     @Override
-    public User saveUser(User User)  {
-
-        String insertQuery = "INSERT INTO login (ID, USERNAME, PASSWORD, ENTITY_ID, DATE_CREATED, DATE_MODIFIED) VALUES (?, ?, ?, ?, ?, ?)";
+    public User saveUser(User user) {
 
         try (Connection connection = ConnectionHelper.getConnection();
```

Comment:
fix the try-with-resource. remove preparedstatement from try()

Reply:
okay napo sir